rightmesh/contracts/Timelock.sol

```diff
@@ -0,0 +1,90 @@
+// Contract to make MeshTokens available to employees with time.
+
+pragma solidity ^0.4.18;
+contract Timelock {
+
+  // todo use SAFEMATH!
+  // todo decimals is a bad idea and won't work - need a way around this should use smallest unit for tokens (wei) and for time seconds is probably alright.
+  // percentages need to be whole numbers... not sure on cleanest way to do this at the moment.
+
+    address public owner; //owner of contract
+
+    uint public cliff_time = 1518696000; //non feb 15 2018
+    uint public cliff_amount = 0.2; //percentage - cliff_amount + gradual_amount = 1 THIS WON'T COMPILE AND DECIMALS ARE A VERY BAD IDEA
+    uint public gradual_amount = 0.8; //percentage - cliff_amount + gradual_amount = 1 THIS WON'T COMPILE AND DECIMALS ARE A VERY BAD IDEA
+    uint public timelock_end = 1525132800; //may 1 00:00:00 2018
+
+    uint public rate_per_second;
+
+    // allocation
+    mapping (address => uint256) public allocated_tokens;
+    mapping (address => uint256) public withdrawn_tokens;
+    //ability to flag accounts
```

Comment:
Instead of having 3 maps. I'm thinking whether we can "struct" data type.

  struct TokenGrant {
    uint256 withdrawn_tokens;   
    uint256 allocated_tokens;     
    bool withdrawal_allowed;
  } 

  mapping (address => TokenGrant[]) public token_vesting;

Reply:
yeah - let's do this